Common/Render/Text/draw_text_sdl.cpp

```diff
@@ -0,0 +1,476 @@
+#endif
+
+	PrepareFallbackFonts();
+}
+
+TextDrawerSDL::~TextDrawerSDL() {
+	ClearCache();
+	TTF_Quit();
+
+#if defined(USE_SDL2_TTF_FONTCONFIG)
+	FcConfigDestroy(config);
+	FcFini();
+#endif
+}
+
+// If a user complains about missing characters on SDL, re-visit this!
+void TextDrawerSDL::PrepareFallbackFonts() {
+#if defined(USE_SDL2_TTF_FONTCONFIG)
+	FcObjectSet *os = FcObjectSetBuild (FC_FILE, FC_INDEX, (char *) 0);
+
+	FcPattern *names[] = {
+		FcNameParse((const FcChar8 *) "Source Han Sans Medium"),
+		FcNameParse((const FcChar8 *) "Droid Sans Bold"),
+		FcNameParse((const FcChar8 *) "DejaVu Sans Condensed"),
+		FcNameParse((const FcChar8 *) "Noto Sans CJK Medium"),
+		FcNameParse((const FcChar8 *) "Noto Sans Hebrew Medium"),
+		FcNameParse((const FcChar8 *) "Noto Sans Lao Medium"),
+		FcNameParse((const FcChar8 *) "Noto Sans Thai Medium")
+	};
+
+	for (int i = 0; i < ARRAY_SIZE(names); i++) {
+    	FcFontSet *foundFonts = FcFontList(config, names[i], os);
+		
+		for (int j = 0; foundFonts && j < foundFonts->nfont; ++j) {
+			FcPattern* font = foundFonts->fonts[j];
+			FcChar8 *path;
+			int fontIndex;
+
+			if (FcPatternGetInteger(font, FC_INDEX, 0, &fontIndex) != FcResultMatch) {
+				fontIndex = 0; // The 0th face is guaranteed to exist 
+			}
+
+			if (FcPatternGetString(font, FC_FILE, 0, &path) == FcResultMatch) {
+				std::string path_str((const char*)path);
+				fallbackFontPaths_.push_back(std::make_pair(path_str, fontIndex));
+			}
+		}
+
+		if (foundFonts) {
+			FcFontSetDestroy(foundFonts);
+		}
+
+		FcPatternDestroy(names[i]);
+	}
+
+	if (os) {
+		FcObjectSetDestroy(os);
+	}
+#elif PPSSPP_PLATFORM(MAC)
+	const char *fontDirs[] = {
+		"/System/Library/Fonts/",
+		"/System/Library/Fonts/Supplemental/",
+		"/Library/Fonts/"
+	};
+
+	const char *fallbackFonts[] = {
+		"Hiragino Sans GB.ttc",
+		"PingFang.ttc",
+		"PingFang SC.ttc",
+		"PingFang TC.ttc",
+		"ヒラギノ角ゴシック W4.ttc",
+		"AppleGothic.ttf",
+		"Arial Unicode.ttf",
+	};
+
+	for (int i = 0; i < ARRAY_SIZE(fontDirs); i++) {
+		for (int j = 0; j < ARRAY_SIZE(fallbackFonts); j++) {
+			Path fontPath = Path(fontDirs[i]) / fallbackFonts[j];
+
+			if (File::Exists(fontPath)) {
+				TTF_Font *openedFont = TTF_OpenFont(fontPath.ToString().c_str(), 24);
+				int64_t numFaces = TTF_FontFaces(openedFont);
+
+				for (int k = 0; k < numFaces; k++) {
+					TTF_Font *fontFace = TTF_OpenFontIndex(fontPath.ToString().c_str(), 24, k);
+					std::string fontFaceName(TTF_FontFaceStyleName(fontFace));
+					TTF_CloseFont(fontFace);
+
+					if (strstr(fontFaceName.c_str(), "Medium") ||
+						strstr(fontFaceName.c_str(), "Regular"))
+					{
+						fallbackFontPaths_.push_back(std::make_pair(fontPath.ToString(), k));
+						break;
+					}
+				}
+
+				TTF_CloseFont(openedFont);
+			}
+		}	
+	}
+#else
+	// We don't have a fallback font for this platform.
+	// Unsupported characters will be rendered as squares.
+#endif
+}
+
+uint32_t TextDrawerSDL::CheckMissingGlyph(std::string& text) {
```

Comment:
nit: Since you're not changing `text`, prefer to use `const std::string &text` here.

-[Unknown]